src/contactmanagementsoftware/service/ChainOfResponsibility/PersonalFriendsValidator.java

```diff
@@ -0,0 +1,58 @@
+package contactmanagementsoftware.service.ChainOfResponsibility;
+
+import contactmanagementsoftware.entity.Contact;
+
+import java.text.ParseException;
+import java.text.SimpleDateFormat;
+import java.util.Date;
+
+public class PersonalFriendsValidator extends ContactValidator {
+    private static final String CONTACT_TYPE = "class contactmanagementsoftware.PersonalFriends";
+
+    public PersonalFriendsValidator(ContactValidator contactValidator) {
+        super(contactValidator);
+    }
+
+    @Override
+    public boolean validateContact(Contact contact) {
+        if (contact.getClass().toString().equalsIgnoreCase(CONTACT_TYPE)) {
+            validateADate(contact.getADate());
+            validateEvents(contact.getEvents());
+            validateAContext(contact.getAContext());
+            return true;
+        }
+        return super.validateContact(contact);
+    }
+
+    private void validateADate(String aDate) {
+        if (aDate == null || aDate.trim().length() < 1) {
+            throw new IllegalArgumentException("Acquaintance date cannot be empty");
+        }
+        try {
+            Date date1 = new SimpleDateFormat("dd/MM/yyyy").parse(aDate);
+            if (date1.after(new Date())) {
+                throw new IllegalArgumentException("Acquaintance date cannot be later than today");
+            }
+        } catch (ParseException e) {
+            throw new IllegalArgumentException("Please enter acquaintance date in (dd/mm/yyyy) format");
```

Comment:
`dd/mm/yyyy` or `dd/MM/yyyy` format?

they are not the same tho.
since line 32, you use `dd/MM/yyyy` can change the error message to that as well?

*Applicable to other error messages as well

Reply:
but for user perspective is dd/mm/yyyy and dd/MM/yyyy diff?

Reply:
ok then